ljepotaservis/ljepotaservis.Web/ClientApp/src/components/superAdminSide/AddStore/AddStore.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from "react";
+import DatePicker from "react-datepicker";
+
+class AddStore extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
```

Comment:
Same as for user this.state = new storeModel

Reply:
This is not a bad idea, but the problem is when the callback onChange to index.js (component above these two) is fired it maps all the values again to index.js and we are left with a regular object again in the state, so I think it would maybe be a good idea to make storeModel and userDto directly in the index.js state so after it is mapped there we are done.

Reply:
This seemed like a good idea to me too and tried to implement it, but since state has to be immutable on every change of the any form value you have to make a new object which is not a problem with anonymous object but with defined objects seems to me like more trouble then it is worth. If you have any ideas on how this could still be done in a simple way I am open to suggestions

Reply:
My idea was use the non anonymous object then act like it's anonymous. This is just state initialization. But if you have issues then it's not worth the hassle.